packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2207,6 +2292,10 @@ export class KeyringController extends BaseController<
     }
 
     this.#keyrings.push(keyring);
+    if (this.#keyringsMetadata.length < this.#keyrings.length) {
+      // For some reason, metadata is not always an extensible array, so .push() is not working
```

Comment:
That truly sounds scary 😅 I would mark it as `FIXME`

```suggestion
      // FIXME: For some reason, metadata is not always an extensible array, so .push() is not working
```

I won't block the review because of this, but this might show that there's a hidden problem somewhere else, and that could backfire some days 🙈